### ISSUE TITLE
DESCRIPTION should be mandatory?

### ISSUE THREAD

**Issue by hadley** [ropensci/rrrpkg#2]
Because it includes machine readable information about:
- the title
- the authors
- the code license

**Comment by gaborcsardi** [comment]
Definitely. Without `DESCRIPTION` it is not a package, anyway. :)

**Comment by benmarwick** [comment]
Right, for a package yes, for a research compendium, perhaps not. Perhaps we should clarify it a bit with this? 

> An optional DESCRIPTION file describes the software dependencies of the compendia. When a DESCRIPTION file is included along with the other items above, then the compendium is also a formal R package and you can take advantage of many time-saving tools for package development, testing and sharing.

**Comment by hadley** [comment]
It's ok for a compendia not to be a package, but surely it's highly desirable to have machine readable metadata about authors, licensing etc?

**Comment by jennybc** [comment]
I also see the value in collecting the list of package dependencies, including versions. Although this bit of info in DESCRIPTION could easily get out of date for a project where the package infrastructure was being used mostly as the inspiration for file organization, i.e. no regular use of `R CMD check`.

**Comment by benmarwick** [comment]
I've seen people putting some of that metadata in a README, in an unstructured way. But perhaps this is a point to push for a bit more standardisation. How about something like this, removing 'optional'?

> A **DESCRIPTION** file that gives structured, machine-readable information about the authors, licensing, the software dependencies and other metadata of the compendium. When a DESCRIPTION file is included along with the other items above, then the compendium is also a formal R package and you can take advantage of many time-saving tools for package development, testing and sharing.

**Comment by ablack3** [comment]
The [OHDSI community](https://www.ohdsi.org/) has been using R packages to implement studies for a while and has many such studies in https://github.com/ohdsi-studies. Personally speaking I think the package structure is not the best fit for all studies. The analyst has to be willing to use the debugger to find code errors and step through execution. I find myself working inside a development version of a study package while actually calling functions in the installed version. We have recently started using renv to record dependencies of a study which is great but could conflict with package dependencies in the DESCRIPTION file. We also have the study metadata in structured readme files and the DESCRIPTION file is sometimes [left with boilerplate](https://github.com/ohdsi-studies/TkrPredictSimple/blob/master/DESCRIPTION) perhaps because people find it is redundant.
